astropy/io/fits/hdu/table.py

```diff
@@ -910,16 +910,18 @@ def __init__(
         ver=None,
         character_as_bytes=False,
     ):
-        from astropy.table import Table
+        # stops unnecessary import of astropy.table.Table if data is None
+        if data is not None:
```

Comment:
We can also avoid the import when data is set to `DELAYED`, which happens when reading a file: 
```suggestion
        if data is not None and data is not DELAYED:
```

Reply:
I do not know much about ```DELAYED```. I found out that it is an object of empty class ```_delayed``` declared in ```base.py```. It would be great if could explain more about ```DELAYED``` .

Reply:
It's just a special object to mark the fact that the data has not been read yet.